java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java

```diff
@@ -416,12 +416,12 @@ void commentsBeforeAndAfterKeywords(){
                         "    Given another\n" +
                         "\n" +
                         "  # before Rule\n" +
-                        "  Rule: \n" +
+                        "  Rule:\n" +
                         "  # after Rule\n" +
                         "\n" +
                         "    # after Rule indent\n" +
                         "    # background comment\n" +
-                        "    Background: foo\n" +
+                        "    Background:\n" +
```

Comment:
Please do create a separate test case for this. That the rule in this test case was nameless was merely a coincidence.

I imagine a feature file with nameless `Feature`, `Rule`, `Background` keywords and possibly `Given`, `When`, `Then` keywords should do.

Reply:
Separate case added.

Blank `Given`, `When` and `Then` are not stripped. It looks like a different situation to me. Do you confirm these should be cleaned? (I expect not as some matching regex may break).

Reply:
Nah. Let's leave them out for now. Not the most important case to fix.